packages/client/src/useObservable.ts

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useState } from "react";
+import { Observable } from "rxjs";
+
+// TODO: allow an optional initial value, but return type should remove the undefined if initial value provided
+//       otherwise, best to use `useObservable(...) ?? defaultValue`
+
+export const useObservable = <T>(observable: Observable<T> | null | undefined) => {
+  const [value, setValue] = useState<T | undefined>(undefined);
+  useEffect(() => {
+    if (!observable) return;
+    const subscription = observable.subscribe(setValue);
+    return () => subscription.unsubscribe();
+  }, [observable]);
+  return value;
+};
```

Comment:
this seems to work re: default value
```suggestion
function useObservable<T>(observable: Observable<T>): T | undefined;
function useObservable<T>(observable: Observable<T>, defaultValue: T): T;

function useObservable<T>(observable: Observable<T>, defaultValue?: T) {
  const [value, setValue] = useState<T | undefined>(defaultValue);
  useEffect(() => {
    if (!observable) return;
    const subscription = observable.subscribe(setValue);
    return () => subscription.unsubscribe();
  }, [observable]);
  return value;
}
```

Reply:
Ohh interesting! I was trying to do this all inline, rather than an overloaded function def.

Reply:
Looks like this is more like a type override and TS doesn't give me warnings/hints when writing the function body itself.